File: java_src/Problem007.java
import java.math.*;


class Problem007
{
	public static void main(String[] args) {
		Integer [] simple_numb = new Integer[500000];
		simple_numb[0] = new Integer(2);
		int arrind=0;
		int i=0;
		int lim=10001;
		//for (int i=0; i<500000; i++)
		while(true)
		{
			i++;
			if ((i>0) && (i%2>0))
			{
				for(int k=0; k<=arrind;k++)
				{
					if((i % simple_numb[k] == 0) || (i <= simple_numb[k]))
					{
						break;
					}
					if(k == arrind)
					{
						//System.out.println("arrind : " + arrind + " simple_numb "+ simple_numb[k]);
						arrind++;
						simple_numb[arrind] = new Integer(i);
						break;
					}
				}
			}
			if (arrind==lim-1)
			{
				break;
			}
		}
		System.out.println(" simple_numb "+ simple_numb[lim-1]);

	}
}

File: py_src/Problem007.py
class PrimeNumbers():
	
	def FindPrimeNumber(self, input_data):
		# assert 0 < int(input_data), "Вводимые данные должны быть числами больше 0 числами и целыми!!!"
		simple_numb = [2]
		arrind = 0
		i = 0
		lim = int(input_data)
		
		while(True):
			i += 1
			if (i > 0) and (i%2 > 0):
				for k in range (arrind+1):
					if(i % simple_numb[k] == 0) or (i <= simple_numb[k]):
						break
					if (k == arrind):
						arrind +=1
						simple_numb.append(i)
						break
			if (arrind == lim-1):
				break
		# print(" simple_numb "+ str(simple_numb[lim-1]))
		return	simple_numb[lim-1]
							
if __name__ == "__main__":
	n = PrimeNumbers()
	print(n.FindPrimeNumber(10001))
		